### PR TITLE
artifacts needed for runqueue state spill evaluation

### PR DESCRIPTION
**Pull request by NamiLiy**
Artifacts needed for run queue state spill evaluation

### CODE REVIEW

File: runqueue_state_spill/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Runqueue state spill evaluation
+
+This folder contain necessary components to measure sate spill evaluation presented in figure 5.
```
Comment (kevinaboos) [inline runqueue_state_spill/README.md]: typo

File: runqueue_state_spill/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+## Description
+This folder contain two pre-built Theseus iso images where spill_free.iso is compiled using default settings and spillful.iso is compiled with state spill enabled. Both iso images are modified to run [rq_eval](../../applications/rq_eval) application upon booting. Each image would measure the average time to add and remove 100000 tasks to a runqueue and average time to spawn and cleanup 100 tasks. Each test is run 10 times to generate average values.
```
Comment (kevinaboos) [inline runqueue_state_spill/README.md]: Provide the exact commands (including configuration) you used to generate each image, such that someone can reproduce it. Make sure it's clear which image is which, and how each one was built.

File: runqueue_state_spill/script.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+for core in ${core_count[@]}; do
+    grep "Log Single" ./results/spill_free_${core}.txt >> ./results/spill_free_single_${core}.csv
+    grep "Log Whole" ./results/spill_free_${core}.txt >> ./results/spill_free_whole_${core}.csv
+    grep "Log Single" ./results/spillful_${core}.txt >> ./results/spillful_single_${core}.csv
+    grep "Log Whole" ./results/spillful_${core}.txt >> ./results/spillful_whole_${core}.csv
+done
+
+# Calculate median and standard deviation from the log
+# please install PTable python package to display output if not available"sudo pip3 install PTable"
```
Comment (kevinaboos) [inline runqueue_state_spill/script.sh]: Take care of this in your script: either check for it being available and print an error message up top, or just install it right away.

The person running your script shouldn't have to open up the script in a text editor to use it.

File: runqueue_state_spill/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+[script.sh](./script.sh) contains the top level script to run the process. The script runs the two images on qemu emulator with cores varying from 2 to 72. The script would analyze the serial outputs of multiple runs and output the results in two tables.
+
+## Evaluation Process
+Run [script.sh](./script.sh)
```
Comment (kevinaboos) [inline runqueue_state_spill/README.md]: Here you need to mention your dependencies and how to install them:
python3, PTable, etc

**Review by kevinaboos — CHANGES_REQUESTED** [review]
needs some minor improvements

File: runqueue_state_spill/README.md (hunk 1 above)
Comment (NamiLiy) [reply runqueue_state_spill/README.md]: Thanks. Corrected

File: runqueue_state_spill/README.md (hunk 2 above)
Comment (NamiLiy) [reply runqueue_state_spill/README.md]: Added

File: runqueue_state_spill/script.sh (hunk 3 above)
Comment (NamiLiy) [reply runqueue_state_spill/script.sh]: Added the check upfront

File: runqueue_state_spill/README.md (hunk 4 above)
Comment (NamiLiy) [reply runqueue_state_spill/README.md]: Added

**Pull request merged**